roles/semaphore_deployment/templates/deploy.j2

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+branch=$@
+allowed=^[a-zA-Z0-9._/\-]+$
+
+# Given branch name must be a single string with no spaces,
+# containing only: ".", "-", "/", "_", or alphanumeric characters.
+
+if [[ ! $branch =~ $allowed ]]
+then
+  echo "Invalid branch name given." >&2
+  exit 1
+fi
+
+cd /home/{{ deployment_user }}/ofn-install
```

Comment:
```suggestion
cd "/home/{{ deployment_user }}/ofn-install"
```

Just in case someone chooses a weird username. It's unlikely, but good practice to use quotes.